cmd/snap-gpio-helper/chip.go

```diff
@@ -0,0 +1,118 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2025 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"unsafe"
+
+	"github.com/snapcore/snapd/dirs"
+	"golang.org/x/sys/unix"
+)
+
+type GPIOChardev interface {
```

Comment:
IMO this belongs somewhere in another package that is just imported here.

Reply:
I was originally thinking of moving it to the gadget package, maybe it should go there?

Reply:
Actually thinking again about it, This is not going to get used in other places (at least for now) and moving into another package would unnecessarily fragment it. What do you think?